llvm/lib/Frontend/OpenMP/OMPIRBuilder.cpp

```diff
@@ -6106,6 +6106,147 @@ void OpenMPIRBuilder::loadOffloadInfoMetadata(Module &M) {
   }
 }
 
+OpenMPIRBuilder::InsertPointTy
+OpenMPIRBuilder::createTeams(const LocationDescription &Loc,
+                             BodyGenCallbackTy BodyGenCB) {
```

Comment:
I still don't understand why we now have basically the same logic 3 times, parallel, tasks, teams.

Reply:
IMO this should be a part of the `OutlineInfo` as a field of argument mapping or something. I will try to work on it as an independent PR.

Reply:
Is this a blocker for this patch? I would like to add that separately because I think it would take a bit longer to design and review. Can we please add support for teams with this independent of that improvement for common usage?